test/e2e/snaps/ppom-malicious-erc20-approval.spec.js

```diff
@@ -0,0 +1,592 @@
+        },
+      };
+    });
+
+// Balance of BUSD
+  await mockServer
+    .forPost()
+    .withJsonBodyIncluding({
+      method: 'eth_call',
+      params: [{ to: BUSD_ADDRESS}]
+    })
+    .thenCallback((req) => {
+      return {
+        statusCode: 200,
+        json: {
+          jsonrpc: '2.0',
+          id: req.body.json.id,
+          result: '0x000000000000000000000000000000000000000000000000000000000001ea4c',
+        },
+      };
+    });
+}
+
+/**
+ * Tests various Blockaid PPOM security alerts. Data for the E2E test requests and responses are provided here:
+ *
+ * @see {@link https://wobbly-nutmeg-8a5.notion.site/MM-E2E-Testing-1e51b617f79240a49cd3271565c6e12d}
+ */
+describe('Confirmation Security Alert - Blockaid', function () {
+
+
+  /**
+   * Disclaimer: this test may be missing checks for some reason types. e.g. blur, domain, and failed
+   */
+  it('should show security alerts for malicious requests', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withNetworkController(mainnetProviderConfig)
+          .withPermissionControllerConnectedToTestDapp()
+          .withPreferencesController({
+            securityAlertsEnabled: true,
+          })
+          .build(),
+        defaultGanacheOptions,
+        testSpecificMock: mockInfura,
+        title: this.test.title,
+      },
+
+      async ({ driver }) => {
+        await driver.navigate();
+        await unlockWallet(driver);
+        await openDapp(driver);
+
+        const expectedTitle = 'This is a deceptive request';
+
+        const testMaliciousConfigs = [
+           {
+             btnSelector: '#maliciousApprovalButton',
+            expectedDescription:
+               'If you approve this request, a third party known for scams might take all your assets.',
+             expectedReason: 'approval_farming',
+           },
+        ];
+
+        for (const config of testMaliciousConfigs) {
+          const { expectedDescription, expectedReason, btnSelector } = config;
+
+          // Click TestDapp button to send JSON-RPC request
+          await driver.clickElement(btnSelector);
+
+          // Wait for confirmation pop-up
+          await driver.waitUntilXWindowHandles(3);
+          const windowHandles = await getWindowHandles(driver, 3);
+          await driver.switchToWindowWithTitle('MetaMask Notification');
+
+          const bannerAlert = await driver.findElement(bannerAlertSelector);
+          const bannerAlertText = await bannerAlert.getText();
+
+          assert(
+            bannerAlertText.includes(expectedTitle),
+            `Expected banner alert title: ${expectedTitle} \nExpected reason: ${expectedReason}\n`,
+          );
+          assert(
+            bannerAlertText.includes(expectedDescription),
+            `Expected banner alert description: ${expectedDescription} \nExpected reason: ${expectedReason}\n`,
+          );
+
+          await driver.delay(10000)
```

Comment:
this can be removed, was just for the recording to fully appreciate the screen appearing :rofl: